src/main/java/io/cdap/plugin/google/sheets/sink/GoogleSheetsSinkConfig.java

```diff
@@ -0,0 +1,263 @@
+ */
+public class GoogleSheetsSinkConfig extends GoogleInputSchemaFieldsUsageConfig {
+  public static final String SHEET_NAME_FIELD_NAME = "sheetName";
+  public static final String SPREADSHEET_NAME_FIELD_NAME = "spreadSheetName";
+  public static final String SCHEMA_SPREAD_SHEET_NAME_FIELD_NAME = "schemaSpreadSheetNameFieldName";
+  public static final String SCHEMA_SHEET_NAME_FIELD_NAME = "schemaSheetNameFieldName";
+  public static final String WRITE_SCHEMA_FIELD_NAME = "writeSchema";
+  public static final String MERGE_DATA_CELLS_FIELD_NAME = "mergeDataCells";
+  public static final String MIN_PAGE_EXTENSION_PAGE_FIELD_NAME = "minPageExtensionSize";
+  public static final String THREADS_NUMBER_FIELD_NAME = "threadsNumber";
+  public static final String MAX_BUFFER_SIZE_FIELD_NAME = "maxBufferSize";
+  public static final String RECORDS_QUEUE_LENGTH_FIELD_NAME = "recordsQueueLength";
+  public static final String MAX_FLUSH_INTERVAL_FIELD_NAME = "maxFlushInterval";
+  public static final String FLUSH_EXECUTION_TIMEOUT_FIELD_NAME = "flushExecutionTimeout";
+
+  public static final String TOP_LEVEL_SCHEMA_MESSAGE =
+    "Field '%s' has unsupported schema type '%s' and logical type '%s'.";
+  public static final String TOP_LEVEL_SCHEMA_CORRECTIVE_MESSAGE =
+    "Allowed top level logical types: [%s] and types: [%s].";
+  public static final String ARRAY_NESTED_SCHEMA_MESSAGE =
+    "Array field '%s' has unsupported components' schema type '%s' and logical type '%s'.";
+  public static final String ARRAY_NESTED_SCHEMA_CORRECTIVE_MESSAGE =
+    "Allowed array nested logical types: [%s] and types: [%s].";
+  public static final String RECORD_NESTED_SCHEMA_MESSAGE =
+    "Record '%s' has field '%s' with unsupported schema type '%s' and logical type '%s'";
+  public static final String RECORD_NESTED_SCHEMA_CORRECTIVE_MESSAGE =
+    "Allowed record nested logical types: [%s] and types: [%s]";
+
+  public static final List<Schema.LogicalType> ALLOWED_LOGICAL_TYPES = Arrays.asList(Schema.LogicalType.DATE,
+      Schema.LogicalType.TIME_MILLIS,
+      Schema.LogicalType.TIME_MICROS,
+      Schema.LogicalType.TIMESTAMP_MILLIS,
+      Schema.LogicalType.TIMESTAMP_MICROS,
+      Schema.LogicalType.DECIMAL);
+
+  public static final List<Schema.Type> ALLOWED_TYPES = Arrays.asList(Schema.Type.STRING,
+      Schema.Type.LONG,
+      Schema.Type.INT,
+      Schema.Type.DOUBLE,
+      Schema.Type.FLOAT,
+      Schema.Type.BYTES,
+      Schema.Type.BOOLEAN,
+      Schema.Type.NULL,
+      Schema.Type.ARRAY,
+      Schema.Type.RECORD);
+
+  public static final List<Schema.Type> ALLOWED_NESTED_TYPES = Arrays.asList(Schema.Type.STRING,
+      Schema.Type.LONG,
+      Schema.Type.INT,
+      Schema.Type.DOUBLE,
+      Schema.Type.FLOAT,
+      Schema.Type.BYTES,
+      Schema.Type.BOOLEAN,
+      Schema.Type.NULL);
+
+  @Name(SHEET_NAME_FIELD_NAME)
+  @Description("Default sheet title. Is used when user don't specify schema field as sheet title.")
```

Comment:
update according to the changes in the doc.

Reply:
done